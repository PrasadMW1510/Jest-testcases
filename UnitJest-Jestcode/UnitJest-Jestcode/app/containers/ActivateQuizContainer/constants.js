/*
 *
 * ActivateQuizContainer constants
 *
 */

export const ACTIVATE_QUIZ_LIST = 'app/ActivateQuizContainer/activate_quiz_list';
export const DEFAULT_ACTION = 'app/ActivateQuizContainer/DEFAULT_ACTION';
export const ACTIVATE_QUIZ_LIST_FAILURE = 'app/activateQuizContainer/activate-quiz-list-failure';
export const INITIALIZE_MODAL_FLAG = 'app/initialize-modal-flag';
