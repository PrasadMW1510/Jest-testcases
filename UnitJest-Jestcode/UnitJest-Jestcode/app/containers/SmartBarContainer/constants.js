/*
 *
 * SmartBarContainer constants
 *
 */

// SmartBar related Action constants
export const SCHOOL_SELECTION = 'school_selection';
export const SCHOOL_SELECTION_SUCCESS = 'school_selection_success';
export const SCHOOL_SELECTION_FAILURE = 'school_selection_failure';

export const SCHOOL_REDIRECTION = 'school_redirection';
export const SCHOOL_REDIRECTION_SUCCESS = 'school_redirection_success';
export const SCHOOL_REDIRECTION_FAILURE = 'school_redirection_failure';

export const CLASS_SELECTION = 'class_selection';
export const CLASS_SELECTION_SUCCESS = 'class_selection_success';
export const CLASS_SELECTION_FAILURE = 'class_selection_failure';

export const CLASS_REDIRECTION = 'class_redirection';
export const CLASS_REDIRECTION_SUCCESS = 'class_redirection_success';
export const CLASS_REDIRECTION_FAILURE = 'class_redirection_failure';

export const GROUP_SELECTION = 'group_selection';
export const GROUP_SELECTION_SUCCESS = 'group_selection_success';
export const GROUP_SELECTION_FAILURE = 'group_selection_failure';

export const STUDENT_SELECTION = 'student_selection';
export const STUDENT_SELECTION_SUCCESS = 'student_selection_success';
export const STUDENT_SELECTION_FAILURE = 'student_selection_failure';

export const GRADE_SELECTION = 'grade_selection';
export const GRADE_SELECTION_SUCCESS = 'grade_selection_success';
export const GRADE_SELECTION_FAILURE = 'grade_selection_failure';

export const TEACHER_REDIRECTION = 'teacher_redirection';
export const TEACHER_REDIRECTION_SUCCESS = 'teacher_redirection_success';
export const TEACHER_REDIRECTION_FAILURE = 'teacher_redirection_failure';

export const CLASS_REDIRECTION_IN_GROUP = 'class_redirection_in_group';
export const CLASS_REDIRECTION_IN_GROUP_SUCCESS = 'class_redirection_in_group_success';
export const CLASS_REDIRECTION_IN_GROUP_FAILURE = 'class_redirection_in_group_failure';

export const GRADE_REDIRECTION = 'grade_redirection';
export const GRADE_REDIRECTION_SUCCESS = 'grade_redirection_success';
export const GRADE_REDIRECTION_FAILURE = 'grade_redirection_failure';

export const TEACHER_SELECTION = 'teacher_selection';
export const TEACHER_SELECTION_SUCCESS = 'teacher_selection_success';
export const TEACHER_SELECTION_FAILURE = 'teacher_selection_failure';

export const GROUP_REDIRECTION = 'group_redirection';

export const RESET_SELECTIONS = 'reset_selections';
export const REDIRECTION_COMMON_SGT = 'redirection_common_SGT';

export const RESET_SMARTBAR_CONSTANTS = 'reset_smartbar_constants';
/**
 * Constants for profile selected value
 * @type {string}
 */
export const PROFILE_SELECTED_SCHOOL = 'profile_selected_school';

export const PROFILE_SELECTED_GRADE = 'profile_selected_grade';

export const PROFILE_SELECTED_TEACHER = 'profile_selected_teacher';

export const PROFILE_SELECTED_CLASS = 'profile_selected_class';

export const PROFILE_SELECTED_GROUP = 'profile_selected_group';

export const PROFILE_SELECTED_STUDENT = 'profile_selected_student';

export const PROFILE_SELECTED_RESET = 'profile_selected_reset';

export const SMARTBAR_SELECTED_UPDATE_DATA = 'smartbar_selected_update_data';
