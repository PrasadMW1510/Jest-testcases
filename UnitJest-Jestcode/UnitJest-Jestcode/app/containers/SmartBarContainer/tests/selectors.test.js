import { fromJS } from 'immutable';

import { COHORT_TYPE } from 'containers/App/constants';

import makeSelectSmartBarContainer, * as Selectors from '../selectors';

describe('SmartBar selectors', () => {
  let mockedState = null;

  beforeEach(() => {
    mockedState = fromJS({
      smartBar: {
        selectedSchoolId: 'selectedSchoolId',
        selectedClassId: 'selectedClassId',
        selectedGroupId: 'selectedGroupId',
        selectedStudentId: 'selectedStudentId',
        selectedGradeId: 'selectedGradeId',
        selectedTeacherId: 'selectedTeacherId',
        makeSelectCohortType: 'makeSelectCohortType',
        clickedSchoolId: 'clickedSchoolId',
        clickedClassId: 'clickedClassId',
        clickedGroupId: 'clickedGroupId',
        clickedStudentId: 'clickedStudentId',
        clickedGradeId: 'clickedGradeId',
        clickedTeacherId: 'clickedTeacherId',
        activeSchoolId: 'activeSchoolId',
        activeGradeId: 'activeGradeId',
        activeTeacherId: 'activeTeacherId',
        activeClassId: 'activeClassId',
        activeGroupId: 'activeGroupId',
        activeStudentId: 'activeStudentId',
      },
      global: {
        userData: {
          classes: {
            selectedClassId: { id: 'selectedClassId', display_name: ['Selected Class'] },
          },
          groups: {
            selectedGroupId: { id: 'selectedGroupId', display_name: ['Selected Group'] },
          },
          teachers: {
            selectedTeacherId: {
              id: 'selectedTeacherId',
              first_name: ['Teacher First Name'],
              last_name: ['Teacher Last Name'],
            },
          },
          grades: {
            selectedGradeId: { id: 'selectedGradeId', full_name: ['Grade Full Name'] },
          },
          students: {
            selectedStudentId: {
              id: 'selectedStudentId',
              first_name: ['Student First Name'],
              last_name: ['Student Last Name'],
            },
          },
          schools: {
            selectedSchoolId: { id: 'selectedSchoolId', name: ['Selected School'] },
          },
          login: {
            user_org: ['School'],
            user_org_id: ['loginSchoolId'],
            user_type: ['Teacher'],
            user_id: ['loginTeacherId'],
          },
        },
        profileData: {
          schools: { activeSchoolId: { name: ['Selected School'] } },
          grades: { activeGradeId: { full_name: ['Grade Full Name'] } },
          teachers: {
            activeTeacherId: {
              first_name: ['Teacher First Name'],
              last_name: ['Teacher Last Name'],
            },
          },
          classes: { activeClassId: { display_name: ['Selected Class'] } },
          groups: { activeGroupId: { display_name: ['Selected Group'] } },
          students: {
            activeStudentId: {
              first_name: ['Student First Name'],
              last_name: ['Student Last Name'],
            },
          },
          failure: null,
        },
      },
    });
  });

  it('should consistently return the smartbar state', () => {
    expect(makeSelectSmartBarContainer()(mockedState)).toMatchSnapshot();
  });

  describe('Actual Selections', () => {
    it('should consistently return the school id', () => {
      expect(Selectors.makeSelectSchoolId()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the grade id', () => {
      expect(Selectors.makeSelectGradeId()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the teacher id', () => {
      expect(Selectors.makeSelectTeacherId()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the group id', () => {
      expect(Selectors.makeSelectGroupId()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the student id', () => {
      expect(Selectors.makeSelectStudentId()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the class id', () => {
      expect(Selectors.makeSelectClassId()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the selected school', () => {
      expect(Selectors.makeSelectSelectedSchoolData()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the selected grade', () => {
      expect(Selectors.makeSelectSelectedGradeData()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the selected teacher', () => {
      expect(Selectors.makeSelectSelectedTeacherData()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the selected group', () => {
      expect(Selectors.makeSelectSelectedGroupData()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the selected student', () => {
      expect(Selectors.makeSelectSelectedStudentData()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the active schoolId', () => {
      expect(Selectors.makeSelectedActiveSchoolId()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the active gradeId', () => {
      expect(Selectors.makeSelectedActiveGradeId()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the active teacherId', () => {
      expect(Selectors.makeSelectedActiveTeacherId()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the active classId', () => {
      expect(Selectors.makeSelectedActiveClassId()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the active groupId', () => {
      expect(Selectors.makeSelectedActiveGroupId()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the active studentId', () => {
      expect(Selectors.makeSelectedActiveStudentId()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the selected class', () => {
      expect(Selectors.makeSelectSelectedClassData()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the selected cohort type', () => {
      expect(Selectors.makeSelectCohortType()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return nothing if there is no selected school', () => {
      mockedState = mockedState.setIn(['global', 'userData'], null);
      expect(Selectors.makeSelectSelectedSchoolData()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return nothing if there is no selected grade', () => {
      mockedState = mockedState.setIn(['global', 'userData'], null);
      expect(Selectors.makeSelectSelectedGradeData()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return nothing if there is no selected teacher', () => {
      mockedState = mockedState.setIn(['global', 'userData'], null);
      expect(Selectors.makeSelectSelectedTeacherData()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return nothing if there is no selected group', () => {
      mockedState = mockedState.setIn(['global', 'userData'], null);
      expect(Selectors.makeSelectSelectedGroupData()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return nothing if there is no selected student', () => {
      mockedState = mockedState.setIn(['global', 'userData'], null);
      expect(Selectors.makeSelectSelectedStudentData()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return nothing if there is no selected class', () => {
      mockedState = mockedState.setIn(['global', 'userData'], null);
      expect(Selectors.makeSelectSelectedClassData()(mockedState)).toMatchSnapshot();
    });
  });

  describe('Clicked Selections', () => {
    it('should consistently return the school id', () => {
      expect(Selectors.makeSelectClickedSchoolId()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the grade id', () => {
      expect(Selectors.makeSelectClickedGradeId()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the teacher id', () => {
      expect(Selectors.makeSelectClickedTeacherId()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the group id', () => {
      expect(Selectors.makeSelectClickedGroupId()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the student id', () => {
      expect(Selectors.makeSelectClickedStudentId()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the class id', () => {
      expect(Selectors.makeSelectClickedClassId()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the selected school', () => {
      expect(Selectors.makeSelectClickedSchoolData()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the selected grade', () => {
      expect(Selectors.makeSelectClickedGradeData()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the selected teacher', () => {
      expect(Selectors.makeSelectClickedTeacherData()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the selected group', () => {
      expect(Selectors.makeSelectClickedGroupData()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the selected student', () => {
      expect(Selectors.makeSelectClickedStudentData()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the selected class', () => {
      expect(Selectors.makeSelectClickedClassData()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return nothing if there is no selected school', () => {
      mockedState = mockedState.setIn(['global', 'userData'], null);
      expect(Selectors.makeSelectClickedSchoolData()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return nothing if there is no selected grade', () => {
      mockedState = mockedState.setIn(['global', 'userData'], null);
      expect(Selectors.makeSelectClickedGradeData()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return nothing if there is no selected teacher', () => {
      mockedState = mockedState.setIn(['global', 'userData'], null);
      expect(Selectors.makeSelectClickedTeacherData()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return nothing if there is no selected group', () => {
      mockedState = mockedState.setIn(['global', 'userData'], null);
      expect(Selectors.makeSelectClickedGroupData()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return nothing if there is no selected student', () => {
      mockedState = mockedState.setIn(['global', 'userData'], null);
      expect(Selectors.makeSelectClickedStudentData()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return nothing if there is no selected class', () => {
      mockedState = mockedState.setIn(['global', 'userData'], null);
      expect(Selectors.makeSelectClickedClassData()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the selectedCohortTypeCohortId for school', () => {
      mockedState = mockedState.setIn(['smartBar', 'selectedCohType'], 'School');
      expect(Selectors.makeSelectCohortTypeCohortId()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the selectedCohortTypeCohortId for grade', () => {
      mockedState = mockedState.setIn(['smartBar', 'selectedCohType'], 'Grade');
      expect(Selectors.makeSelectCohortTypeCohortId()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the selectedCohortTypeCohortId for teacher', () => {
      mockedState = mockedState.setIn(['smartBar', 'selectedCohType'], 'Teacher');
      expect(Selectors.makeSelectCohortTypeCohortId()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the selectedCohortTypeCohortId for class', () => {
      mockedState = mockedState.setIn(['smartBar', 'selectedCohType'], 'Class');
      expect(Selectors.makeSelectCohortTypeCohortId()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the selectedCohortTypeCohortId for group', () => {
      mockedState = mockedState.setIn(['smartBar', 'selectedCohType'], 'Group');
      expect(Selectors.makeSelectCohortTypeCohortId()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the selectedCohortTypeCohortId for student', () => {
      mockedState = mockedState.setIn(['smartBar', 'selectedCohType'], 'Student');
      expect(Selectors.makeSelectCohortTypeCohortId()(mockedState)).toMatchSnapshot();
    });

    it('should consistently return the selectedCohortTypeCohortId when none selected', () => {
      mockedState = mockedState.setIn(['smartBar', 'selectedCohType'], null);
      expect(Selectors.makeSelectCohortTypeCohortId()(mockedState)).toMatchSnapshot();
    });
  });
  describe('cohort selections', () => {
    it('should consistently return the selectedEffectiveCohortObject when none selected and District userOrg', () => {
      mockedState = mockedState.setIn(['smartBar', 'selectedCohType'], null);
      mockedState = mockedState.setIn(['global', 'userData', 'login', 'user_org', 0], 'District');
      mockedState = mockedState.setIn(
        ['global', 'userData', 'login', 'user_org_id', 0],
        'loginDistrictId'
      );
      const result = Selectors.makeSelectEffectiveCohortObject()(mockedState);
      expect(result).toMatchSnapshot();
      expect(result.cohortType).toEqual(COHORT_TYPE.District);
      expect(result.id).toEqual('loginDistrictId');
    });

    it('should consistently return the selectedEffectiveCohortObject when none selected and School userOrg', () => {
      mockedState = mockedState.setIn(['smartBar', 'selectedCohType'], null);
      const result = Selectors.makeSelectEffectiveCohortObject()(mockedState);
      expect(result).toMatchSnapshot();
      expect(result.cohortType).toEqual(COHORT_TYPE.School);
      expect(result.id).toEqual('loginSchoolId');
    });

    it('should consistently return the selectedEffectiveCohortObject when none selected and Teacher is logged in', () => {
      mockedState = mockedState.setIn(['smartBar', 'selectedCohType'], null);
      mockedState = mockedState.setIn(['global', 'userData', 'login', 'user_org'], null);
      mockedState = mockedState.setIn(['global', 'userData', 'login', 'user_org_id'], null);
      const result = Selectors.makeSelectEffectiveCohortObject()(mockedState);
      expect(result).toMatchSnapshot();
      expect(result.cohortType).toEqual(COHORT_TYPE.Teacher);
      expect(result.id).toEqual('loginTeacherId');
    });
    it('should consistently return the selectedEffectiveCohortObject when School selected in SmartBar', () => {
      mockedState = mockedState.setIn(['smartBar', 'selectedCohType'], 'School');
      const result = Selectors.makeSelectEffectiveCohortObject()(mockedState);
      expect(result).toMatchSnapshot();
      expect(result.cohortType).toEqual(COHORT_TYPE.School);
      expect(result.id).toEqual('selectedSchoolId');
    });

    it('should consistently return the selectedEffectiveCohortObject when Grade selected in SmartBar', () => {
      mockedState = mockedState.setIn(['smartBar', 'selectedCohType'], 'Grade');
      const result = Selectors.makeSelectEffectiveCohortObject()(mockedState);
      expect(result).toMatchSnapshot();
      expect(result.cohortType).toEqual(COHORT_TYPE.Grade);
      expect(result.id).toEqual('selectedGradeId');
      expect(result.schoolId).toEqual('selectedSchoolId');
    });

    it('should consistently return selected grade with school when user type is DA/DTA', () => {
      mockedState = mockedState.setIn(['smartBar', 'selectedCohType'], 'Grade');
      mockedState = mockedState.setIn(['smartBar', 'selectedSchoolId'], null);
      const result = Selectors.makeSelectEffectiveCohortObject()(mockedState);
      expect(result.cohortType).toEqual(COHORT_TYPE.Grade);
      expect(result.schoolId).toEqual('activeSchoolId');
      expect(result.id).toEqual('selectedGradeId');
    });

    it('should consistently return the selectedEffectiveCohortObject when teacher selected in SmartBar', () => {
      mockedState = mockedState.setIn(['smartBar', 'selectedCohType'], 'Teacher');
      const result = Selectors.makeSelectEffectiveCohortObject()(mockedState);
      expect(result).toMatchSnapshot();
      expect(result.cohortType).toEqual(COHORT_TYPE.Teacher);
      expect(result.id).toEqual('selectedTeacherId');
    });

    it('should consistently return the selectedEffectiveCohortObject when Class selected in SmartBar', () => {
      mockedState = mockedState.setIn(['smartBar', 'selectedCohType'], 'Class');
      const result = Selectors.makeSelectEffectiveCohortObject()(mockedState);
      expect(result).toMatchSnapshot();
      expect(result.cohortType).toEqual(COHORT_TYPE.Class);
      expect(result.id).toEqual('selectedClassId');
    });

    it('should consistently return the selectedEffectiveCohortObject when Group selected in SmartBar', () => {
      mockedState = mockedState.setIn(['smartBar', 'selectedCohType'], 'Group');
      const result = Selectors.makeSelectEffectiveCohortObject()(mockedState);
      expect(result).toMatchSnapshot();
      expect(result.cohortType).toEqual(COHORT_TYPE.Group);
      expect(result.id).toEqual('selectedGroupId');
    });

    it('should consistently return the selectedEffectiveCohortObject when Student selected in SmartBar', () => {
      mockedState = mockedState.setIn(['smartBar', 'selectedCohType'], 'Student');
      const result = Selectors.makeSelectEffectiveCohortObject()(mockedState);
      expect(result).toMatchSnapshot();
      expect(result.cohortType).toEqual(COHORT_TYPE.Student);
      expect(result.id).toEqual('selectedStudentId');
    });
  });

  describe('makeSelectSelectedCohortName', () => {
    it('should return correct value for selected school', () => {
      mockedState = mockedState.setIn(['smartBar', 'selectedCohType'], 'School');
      expect(Selectors.makeSelectSelectedCohortName()(mockedState)).toEqual('Selected School');
    });

    it('should return correct value for selected student', () => {
      mockedState = mockedState.setIn(['smartBar', 'selectedCohType'], 'Student');
      expect(Selectors.makeSelectSelectedCohortName()(mockedState)).toEqual(
        'Student Last Name, Student First Name'
      );
    });

    it('should return correct value for selected grade', () => {
      mockedState = mockedState.setIn(['smartBar', 'selectedCohType'], 'Grade');
      expect(Selectors.makeSelectSelectedCohortName()(mockedState)).toEqual('Grade Full Name');
    });

    it('should return correct value for selected teacher', () => {
      mockedState = mockedState.setIn(['smartBar', 'selectedCohType'], 'Teacher');
      expect(Selectors.makeSelectSelectedCohortName()(mockedState)).toEqual(
        'Teacher Last Name, Teacher First Name'
      );
    });

    it('should return correct value for selected class', () => {
      mockedState = mockedState.setIn(['smartBar', 'selectedCohType'], 'Class');
      expect(Selectors.makeSelectSelectedCohortName()(mockedState)).toEqual('Selected Class');
    });

    it('should return correct value for selected group', () => {
      mockedState = mockedState.setIn(['smartBar', 'selectedCohType'], 'Group');
      expect(Selectors.makeSelectSelectedCohortName()(mockedState)).toEqual('Selected Group');
    });

    it('should return correct value for no data available', () => {
      mockedState = mockedState.setIn(['smartBar', 'selectedCohType'], 'Teacher');
      mockedState = mockedState.setIn(['global', 'profileData', 'teachers'], null);
      expect(Selectors.makeSelectSelectedCohortName()(mockedState)).toEqual('');
    });
  });

  describe('makeSelectActiveSchoolId', () => {
    it('should consistently return school ID for SA users', () => {
      mockedState = mockedState.setIn(['smartBar', 'selectedSchoolId'], null);
      mockedState = mockedState.setIn(
        ['global', 'userData', 'profile', 'organizations', 0, 'organization', 0, 'type', 0],
        'School'
      );
      mockedState = mockedState.setIn(
        ['global', 'userData', 'profile', 'organizations', 0, 'organization', 0, 'org_id', 0],
        'schoolId'
      );
      const schoolId = Selectors.makeSelectActiveSchoolId()(mockedState);
      expect(schoolId).toEqual('schoolId');
    });
  });
});
