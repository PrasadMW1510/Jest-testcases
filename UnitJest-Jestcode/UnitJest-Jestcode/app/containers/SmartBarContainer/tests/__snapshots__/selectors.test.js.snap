// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmartBar selectors Actual Selections should consistently return nothing if there is no selected class 1`] = `
Immutable.Map {
  display_name: Immutable.List [
    "Selected Class",
  ],
}
`;

exports[`SmartBar selectors Actual Selections should consistently return nothing if there is no selected grade 1`] = `
Immutable.Map {
  full_name: Immutable.List [
    "Grade Full Name",
  ],
}
`;

exports[`SmartBar selectors Actual Selections should consistently return nothing if there is no selected group 1`] = `
Immutable.Map {
  display_name: Immutable.List [
    "Selected Group",
  ],
}
`;

exports[`SmartBar selectors Actual Selections should consistently return nothing if there is no selected school 1`] = `
Immutable.Map {
  name: Immutable.List [
    "Selected School",
  ],
}
`;

exports[`SmartBar selectors Actual Selections should consistently return nothing if there is no selected student 1`] = `
Immutable.Map {
  first_name: Immutable.List [
    "Student First Name",
  ],
  last_name: Immutable.List [
    "Student Last Name",
  ],
}
`;

exports[`SmartBar selectors Actual Selections should consistently return nothing if there is no selected teacher 1`] = `
Immutable.Map {
  first_name: Immutable.List [
    "Teacher First Name",
  ],
  last_name: Immutable.List [
    "Teacher Last Name",
  ],
}
`;

exports[`SmartBar selectors Actual Selections should consistently return the active classId 1`] = `"activeClassId"`;

exports[`SmartBar selectors Actual Selections should consistently return the active gradeId 1`] = `"activeGradeId"`;

exports[`SmartBar selectors Actual Selections should consistently return the active groupId 1`] = `"activeGroupId"`;

exports[`SmartBar selectors Actual Selections should consistently return the active schoolId 1`] = `"activeSchoolId"`;

exports[`SmartBar selectors Actual Selections should consistently return the active studentId 1`] = `"activeStudentId"`;

exports[`SmartBar selectors Actual Selections should consistently return the active teacherId 1`] = `"activeTeacherId"`;

exports[`SmartBar selectors Actual Selections should consistently return the class id 1`] = `"selectedClassId"`;

exports[`SmartBar selectors Actual Selections should consistently return the grade id 1`] = `"selectedGradeId"`;

exports[`SmartBar selectors Actual Selections should consistently return the group id 1`] = `"selectedGroupId"`;

exports[`SmartBar selectors Actual Selections should consistently return the school id 1`] = `"selectedSchoolId"`;

exports[`SmartBar selectors Actual Selections should consistently return the selected class 1`] = `
Immutable.Map {
  display_name: Immutable.List [
    "Selected Class",
  ],
}
`;

exports[`SmartBar selectors Actual Selections should consistently return the selected cohort type 1`] = `undefined`;

exports[`SmartBar selectors Actual Selections should consistently return the selected grade 1`] = `
Immutable.Map {
  full_name: Immutable.List [
    "Grade Full Name",
  ],
}
`;

exports[`SmartBar selectors Actual Selections should consistently return the selected group 1`] = `
Immutable.Map {
  display_name: Immutable.List [
    "Selected Group",
  ],
}
`;

exports[`SmartBar selectors Actual Selections should consistently return the selected school 1`] = `
Immutable.Map {
  name: Immutable.List [
    "Selected School",
  ],
}
`;

exports[`SmartBar selectors Actual Selections should consistently return the selected student 1`] = `
Immutable.Map {
  first_name: Immutable.List [
    "Student First Name",
  ],
  last_name: Immutable.List [
    "Student Last Name",
  ],
}
`;

exports[`SmartBar selectors Actual Selections should consistently return the selected teacher 1`] = `
Immutable.Map {
  first_name: Immutable.List [
    "Teacher First Name",
  ],
  last_name: Immutable.List [
    "Teacher Last Name",
  ],
}
`;

exports[`SmartBar selectors Actual Selections should consistently return the student id 1`] = `"selectedStudentId"`;

exports[`SmartBar selectors Actual Selections should consistently return the teacher id 1`] = `"selectedTeacherId"`;

exports[`SmartBar selectors Clicked Selections should consistently return nothing if there is no selected class 1`] = `null`;

exports[`SmartBar selectors Clicked Selections should consistently return nothing if there is no selected grade 1`] = `null`;

exports[`SmartBar selectors Clicked Selections should consistently return nothing if there is no selected group 1`] = `null`;

exports[`SmartBar selectors Clicked Selections should consistently return nothing if there is no selected school 1`] = `null`;

exports[`SmartBar selectors Clicked Selections should consistently return nothing if there is no selected student 1`] = `null`;

exports[`SmartBar selectors Clicked Selections should consistently return nothing if there is no selected teacher 1`] = `null`;

exports[`SmartBar selectors Clicked Selections should consistently return the class id 1`] = `"clickedClassId"`;

exports[`SmartBar selectors Clicked Selections should consistently return the grade id 1`] = `"clickedGradeId"`;

exports[`SmartBar selectors Clicked Selections should consistently return the group id 1`] = `"clickedGroupId"`;

exports[`SmartBar selectors Clicked Selections should consistently return the school id 1`] = `"clickedSchoolId"`;

exports[`SmartBar selectors Clicked Selections should consistently return the selected class 1`] = `undefined`;

exports[`SmartBar selectors Clicked Selections should consistently return the selected grade 1`] = `undefined`;

exports[`SmartBar selectors Clicked Selections should consistently return the selected group 1`] = `undefined`;

exports[`SmartBar selectors Clicked Selections should consistently return the selected school 1`] = `undefined`;

exports[`SmartBar selectors Clicked Selections should consistently return the selected student 1`] = `undefined`;

exports[`SmartBar selectors Clicked Selections should consistently return the selected teacher 1`] = `undefined`;

exports[`SmartBar selectors Clicked Selections should consistently return the selectedCohortTypeCohortId for class 1`] = `
Object {
  "id": "selectedClassId",
  "selectedCohortType": "Class",
}
`;

exports[`SmartBar selectors Clicked Selections should consistently return the selectedCohortTypeCohortId for grade 1`] = `
Object {
  "id": "selectedGradeId",
  "selectedCohortType": "Grade",
}
`;

exports[`SmartBar selectors Clicked Selections should consistently return the selectedCohortTypeCohortId for group 1`] = `
Object {
  "id": "selectedGroupId",
  "selectedCohortType": "Group",
}
`;

exports[`SmartBar selectors Clicked Selections should consistently return the selectedCohortTypeCohortId for school 1`] = `
Object {
  "id": "selectedSchoolId",
  "selectedCohortType": "School",
}
`;

exports[`SmartBar selectors Clicked Selections should consistently return the selectedCohortTypeCohortId for student 1`] = `
Object {
  "id": "selectedStudentId",
  "selectedCohortType": "Student",
}
`;

exports[`SmartBar selectors Clicked Selections should consistently return the selectedCohortTypeCohortId for teacher 1`] = `
Object {
  "id": "selectedTeacherId",
  "selectedCohortType": "Teacher",
}
`;

exports[`SmartBar selectors Clicked Selections should consistently return the selectedCohortTypeCohortId when none selected 1`] = `Object {}`;

exports[`SmartBar selectors Clicked Selections should consistently return the student id 1`] = `"clickedStudentId"`;

exports[`SmartBar selectors Clicked Selections should consistently return the teacher id 1`] = `"clickedTeacherId"`;

exports[`SmartBar selectors cohort selections should consistently return the selectedEffectiveCohortObject when Class selected in SmartBar 1`] = `
Object {
  "cohortType": "Class",
  "id": "selectedClassId",
  "schoolId": "selectedSchoolId",
}
`;

exports[`SmartBar selectors cohort selections should consistently return the selectedEffectiveCohortObject when Grade selected in SmartBar 1`] = `
Object {
  "cohortType": "Grade",
  "id": "selectedGradeId",
  "schoolId": "selectedSchoolId",
}
`;

exports[`SmartBar selectors cohort selections should consistently return the selectedEffectiveCohortObject when Group selected in SmartBar 1`] = `
Object {
  "cohortType": "Group",
  "id": "selectedGroupId",
  "schoolId": "selectedSchoolId",
}
`;

exports[`SmartBar selectors cohort selections should consistently return the selectedEffectiveCohortObject when School selected in SmartBar 1`] = `
Object {
  "cohortType": "School",
  "id": "selectedSchoolId",
}
`;

exports[`SmartBar selectors cohort selections should consistently return the selectedEffectiveCohortObject when Student selected in SmartBar 1`] = `
Object {
  "cohortType": "Student",
  "id": "selectedStudentId",
  "schoolId": "selectedSchoolId",
}
`;

exports[`SmartBar selectors cohort selections should consistently return the selectedEffectiveCohortObject when none selected and District userOrg 1`] = `
Object {
  "cohortType": "District",
  "id": "loginDistrictId",
}
`;

exports[`SmartBar selectors cohort selections should consistently return the selectedEffectiveCohortObject when none selected and School userOrg 1`] = `
Object {
  "cohortType": "School",
  "id": "loginSchoolId",
}
`;

exports[`SmartBar selectors cohort selections should consistently return the selectedEffectiveCohortObject when none selected and Teacher is logged in 1`] = `
Object {
  "cohortType": "Teacher",
  "id": "loginTeacherId",
  "schoolId": "selectedSchoolId",
}
`;

exports[`SmartBar selectors cohort selections should consistently return the selectedEffectiveCohortObject when teacher selected in SmartBar 1`] = `
Object {
  "cohortType": "Teacher",
  "id": "selectedTeacherId",
  "schoolId": "selectedSchoolId",
}
`;

exports[`SmartBar selectors should consistently return the smartbar state 1`] = `
Immutable.Map {
  selectedTeacherId: "selectedTeacherId",
  selectedSchoolId: "selectedSchoolId",
  clickedGradeId: "clickedGradeId",
  activeGradeId: "activeGradeId",
  clickedClassId: "clickedClassId",
  activeClassId: "activeClassId",
  clickedStudentId: "clickedStudentId",
  activeStudentId: "activeStudentId",
  clickedGroupId: "clickedGroupId",
  activeGroupId: "activeGroupId",
  activeSchoolId: "activeSchoolId",
  clickedTeacherId: "clickedTeacherId",
  clickedSchoolId: "clickedSchoolId",
  activeTeacherId: "activeTeacherId",
  selectedGradeId: "selectedGradeId",
  selectedClassId: "selectedClassId",
  makeSelectCohortType: "makeSelectCohortType",
  selectedStudentId: "selectedStudentId",
  selectedGroupId: "selectedGroupId",
}
`;
